mp/src/game/client/momentum/c_mom_player.cpp

```diff
@@ -86,6 +91,22 @@ bool C_MomentumPlayer::CreateMove(float flInputSampleTime, CUserCmd *pCmd)
     return BaseClass::CreateMove(flInputSampleTime, pCmd);
 }
 
+void C_MomentumPlayer::Spawn()
+{
+    C_BasePlayer::Spawn();
+
+    SetNextClientThink(CLIENT_THINK_ALWAYS);
+}
+
+void C_MomentumPlayer::ClientThink()
+{
+    SetNextClientThink(CLIENT_THINK_ALWAYS);
+    if (m_pLeaderboard->IsVisible())
+        m_Local.m_iHideHUD |= HIDEHUD_LEADERBOARDS;
+    else
+        m_Local.m_iHideHUD &= ~HIDEHUD_LEADERBOARDS;
```

Comment:
I'm thinking that maybe it's better to call a private function to set this from the leaderboards class itself, when its visibility is toggled. Since `m_Local` is public you probably could just make it a function inside the leaderboards class like `ToggleLeaderboardsHidehud(book bVisible)` and call this code inside it.

You can get the player with `C_BasePlayer::GetLocalPlayer();`